HW2/P3/mandelbrot.pyx

```diff
@@ -30,19 +35,58 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
     assert in_coords.shape[1] % 8 == 0, "Input array must have 8N columns"
     assert in_coords.shape[0] == out_counts.shape[0], "Input and output arrays must be the same size"
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
-
-    with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
+    
+    out_counts2 = np.copy(out_counts)
+    num_threads = 1
+    thresh = AVX.float_to_float8(4.0)
+    one8 = AVX.make_float8(1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0)
+    re = np.real(in_coords)
+    im = np.imag(in_coords)
+    print "num threads: ",num_threads
+    with nogil, parallel(num_threads=num_threads):
+        val0 = threadid()
+        for i in prange(in_coords.shape[0], schedule='static', chunksize = 1):
+            for j in range(in_coords.shape[1]/8):
+                cRe = AVX.make_float8(re[i, 8*j],re[i, 8*j+1],re[i, 8*j+2],re[i, 8*j+3],re[i, 8*j+4],re[i, 8*j+5],re[i, 8*j+6],re[i, 8*j+7])
+                cIm = AVX.make_float8(im[i, 8*j],im[i, 8*j+1],im[i, 8*j+2],im[i, 8*j+3],im[i, 8*j+4],im[i, 8*j+5],im[i, 8*j+6],im[i, 8*j+7])
```

Comment:
I think you could make this code easier to read if you made the "for j in range(...)" part have a step size of instead of multiplying by 8 every time. Also, splitting the cRe and cIm into multiple lines would help